test/list_notifier_test.dart

```diff
@@ -0,0 +1,157 @@
+import 'package:flutter_test/flutter_test.dart';
+
+import '../lib/listenable_collections.dart';
+
+void main() {
+  ListNotifier list;
+
+  setUp(() {
+    list = ListNotifier(data: [1, 2, 3]);
+  });
+
+
+  test("Elements get swapped correctly", () {
+    list.swap(0, 2);
+
+    expect(list.value, [3, 2, 1]);
+  });
+
+  test("Listeners get updated if a value gets added", () {
+    list.addListener(() {
+      expect(list[3], 4);
+    });
+
+    list.add(4);
+  });
+
+  test("Listeners get notified if an iterable is added", () {
+    list.addListener(() {
```

Comment:
Sorry to correct you here.
if the Listener isn't called `expect` won't be called ever.
So the right way to do it is to assign some value inside the handler and check that this assignment with an expect

Reply:
this applies to all the following tests

Reply:
No, you are perfectly right, my bad, I'll fix it later today

Reply:
I bumped into a problem, the listener isn't called immediately, therefore the following test will fail because result is empty. Do you have any suggestion?

`  test("Listeners get notified on swap", () async {
    list.addListener(() {
      result = list.value;
    });

    list.swap(0, 2);

    expect(result, [3, 2, 1]);
  });`